File: example/java/jimmer-sql-graphql/src/main/java/org/babyfish/jimmer/sql/example/graphql/input/AuthorInput.java
package org.babyfish.jimmer.sql.example.graphql.entities.input;

import org.babyfish.jimmer.sql.example.graphql.entities.Author;
import org.babyfish.jimmer.sql.example.graphql.entities.AuthorDraft;
import org.babyfish.jimmer.sql.example.graphql.entities.Gender;
import org.springframework.lang.Nullable;

public class AuthorInput {

    @Nullable
    private final Long id;

    private final String firstName;

    private final String lastName;

    private final Gender gender;

    public AuthorInput(
            @Nullable Long id,
            String firstName,
            String lastName,
            Gender gender
    ) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.gender = gender;
    }

    public Author toAuthor() {
        return AuthorDraft.$.produce(author -> {
            if (id != null) {
                author.setId(id);
            }
            author
                    .setFirstName(firstName)
                    .setLastName(lastName)
                    .setGender(gender);
        });
    }
}

File: example/kotlin/jimmer-sql-graphql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/graphql/input/AuthorInput.kt
package org.babyfish.jimmer.example.kt.graphql.input

import org.babyfish.jimmer.example.kt.graphql.entities.Author
import org.babyfish.jimmer.example.kt.graphql.entities.Gender
import org.babyfish.jimmer.example.kt.graphql.entities.by
import org.babyfish.jimmer.kt.new

class AuthorInput(
    val id: Long?, // Optional id
    val firstName: String,
    val lastName: String,
    val gender: Gender
) {

    fun toAuthor(): Author =
        new(Author::class).by {
            val that = this@AuthorInput
            that.id?.let {
                id = it
            }
            firstName = that.firstName
            lastName = that.lastName
            gender = that.gender
        }
}